vizro-core/src/vizro/models/_navigation/accordion.py

```diff
@@ -79,22 +79,17 @@ def build(self, *, active_page_id=None):
             id="nav-panel",
         )
 
-    def _create_nav_links(self, pages):
-        """Creates a `NavLink` for each provided page that is registered."""
+    def _create_nav_links(self, pages: List[str]):
+        """Creates a `NavLink` for each provided page."""
         nav_links = []
+
         for page_id in pages:
-            try:
-                page = dash.page_registry[page_id]
-            except KeyError as exc:
-                raise KeyError(
-                    f"Page with ID {page_id} cannot be found. Please add the page to `Dashboard.pages`"
-                ) from exc
             nav_links.append(
                 dbc.NavLink(
-                    children=[page["name"]],
+                    children=page_id,
                     className="accordion-item-link",
                     active="exact",
-                    href=page["relative_path"],
+                    href=model_manager[ModelID(str(page_id))].path,
```

Comment:
@antonymilne - had to change this one to take the `page.path` from the model_manager in case someone sets a different `page.path` than the` page.id`